backend/services/users.js

```diff
@@ -113,6 +114,12 @@ async function editUser(rawUser, editingUser) {
       continue;
     }
     if (editableFields.has(field)) {
+      if (field === "role" && editedUser.role.name === "Pending") {
+        throw ServiceError(403, "You cannot change the role of a Pending user.");
+      }
+      if (field === "role" && newValue === pending_role._id.toString()) {
```

Comment:
Instead of finding the Pending role and checking whether the new role ID matches the pending role's ID, can we find the Role document corresponding to `rawUser.role` and check whether that role is the pending role? That should also handle the case where `rawUser.role` doesn't correspond to a valid Role document.

Reply:
ah yea good point!